dlsonic/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+cmake_minimum_required(VERSION 3.13.0)
+project(dlsonicproject LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
```

Comment:
If you have time, turn on additional warnings (in the testbins also) and clean up the code.

```
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
```

Reply:
This one remains, but its coming out of dyninst that I haven't updated from a while. I remember you had fixed a bunch of those in dyninst already so simply updating dyninst will fix it. Will give it a try later after other comments.

```shell
/p/paradyn/development/adityajain/props/builddir/lib/cmake/Dyninst/../../../include/tbb/concurrent_hash_map.h:122:24: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘class tbb::interface5::internal::hash_map_base’ with no trivial copy-assignment; use value-initialization instead [-Wclass-memaccess]
  122 |             std::memset( this, 0, pointers_per_table*sizeof(segment_ptr_t) // 32*4=128   or 64*8=512
      |             ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  123 |                 + sizeof(my_size) + sizeof(my_mask)  // 4+4 or 8+8
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  124 |                 + embedded_buckets*sizeof(bucket) ); // n*8 or n*16
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/p/paradyn/development/adityajain/props/builddir/lib/cmake/Dyninst/../../../include/tbb/concurrent_hash_map.h:76:11: note: ‘class tbb::interface5::internal::hash_map_base’ declared here
   76 |     class hash_map_base {
      |           ^~~~~~~~~~~~~
```

Reply:
This is a warning in the TBB header. The latest (or soon to be) version of Dyninst makes tbb headers be system headers and warning will appear.